examples/common-switches.mk

```diff
@@ -250,6 +251,15 @@ else
 COMMONCFLAGS += -DOPENTHREAD_SETTINGS_RAM=0
 endif
 
+ifeq ($(OTNS),1)
+
+ifneq ($(VIRTUAL_TIME),1)
+$(error OTNS only supports virtual time simulation.)
```

Comment:
move option dependency check into https://github.com/openthread/openthread/blob/master/src/core/config/openthread-core-config-check.h?

Reply:
There are currently two checks required for OTNS:
- VIRTUAL_TIME must be 1
- VIRTUAL_TIME_UART must be 0

Currently, the default value of `OPENTHREAD_SIMULATION_VIRTUAL_TIME` and `OPENTHREAD_SIMULATION_VIRTUAL_TIME_UART` are both defined in `examples/simulation/platform-config.h`, which however is not included by `openthread-core-config-check.h`. So, we can not check these two MACROs in `openthread-core-config-check.h`. 

So, I put the check in `examples/simulation/platform-config.h`. 

Thoughts?

Reply:
For platform specific options check, one existing example I can find is in the Makefile
https://github.com/openthread/openthread/blob/master/examples/Makefile-nrf52840#L81-L83

Maybe here the check could be put into `Makefile-simulation`?
Besides, `OTNS` seems also a platform specific option - only for simulation, maybe also move to the  Makefile?

Thoughts? @jwhui @simonlingoogle

Reply:
Putting checks to `Makefile-simualtion` sounds good to me.